package kyrinbox.server;

message ChunkStatus {
    required uint32 kbid = 1;
}

message ChunkClusterStatus {
    repeated ChunkStatus seeds            = 1;
    repeated ChunkStatus commons          = 2;
    repeated ChunkStatus dead_seeds       = 3;
    repeated ChunkStatus dead_commons     = 4;
    repeated ChunkClusterGossipData datas = 5;
    optional uint64 timestamp             = 6;
}

message ChunkClusterGossipData {
    required uint32 kbid                    = 1;
    required string host                    = 2;
    required uint32 gossip_port             = 3;
    required uint32 download_chunk_port     = 4;
    required uint32 upload_chunk_port       = 5;
    required uint32 set_file_keyinfo_port   = 6;
    required uint32 get_file_keyinfo_port   = 7;
}
